src/main/java/org/alfresco/repo/search/impl/QueryParserUtils.java

```diff
@@ -384,6 +384,36 @@ else if (field.endsWith(FIELD_SOLR_NOLOCALE_TOKENISED_SUFFIX))
             propertyFieldName = field.substring(field.startsWith("@") ? 1 : 0, field.length() - FIELD_SOLR_NOLOCALE_TOKENISED_SUFFIX.length());
             ending = FIELD_SOLR_NOLOCALE_TOKENISED_SUFFIX;
         }
+        else if (field.endsWith(FIELD_SOLR_UNIT_OF_TIME_SECOND_SUFFIX))
+        {
+            propertyFieldName = field.substring(field.startsWith("@") ? 1 : 0, field.length() - FIELD_SOLR_UNIT_OF_TIME_SECOND_SUFFIX.length());
+            ending = FIELD_SOLR_UNIT_OF_TIME_SECOND_SUFFIX;
+        }
+        else if (field.endsWith(FIELD_SOLR_UNIT_OF_TIME_MINUTE_SUFFIX))
+        {
+            propertyFieldName = field.substring(field.startsWith("@") ? 1 : 0, field.length() - FIELD_SOLR_UNIT_OF_TIME_MINUTE_SUFFIX.length());
+            ending = FIELD_SOLR_UNIT_OF_TIME_MINUTE_SUFFIX;
+        }
+        else if (field.endsWith(FIELD_SOLR_UNIT_OF_TIME_HOUR_SUFFIX))
+        {
+            propertyFieldName = field.substring(field.startsWith("@") ? 1 : 0, field.length() - FIELD_SOLR_UNIT_OF_TIME_HOUR_SUFFIX.length());
+            ending = FIELD_SOLR_UNIT_OF_TIME_HOUR_SUFFIX;
+        }
+        else if (field.endsWith(FIELD_SOLR_UNIT_OF_TIME_DAY_SUFFIX))
+        {
+            propertyFieldName = field.substring(field.startsWith("@") ? 1 : 0, field.length() - FIELD_SOLR_UNIT_OF_TIME_DAY_SUFFIX.length());
+            ending = FIELD_SOLR_UNIT_OF_TIME_DAY_SUFFIX;
+        }
+        else if (field.endsWith(FIELD_SOLR_UNIT_OF_TIME_MONTH_SUFFIX))
+        {
+            propertyFieldName = field.substring(field.startsWith("@") ? 1 : 0, field.length() - FIELD_SOLR_UNIT_OF_TIME_MONTH_SUFFIX.length());
+            ending = FIELD_SOLR_UNIT_OF_TIME_MONTH_SUFFIX;
+        }
+        else if (field.endsWith(FIELD_SOLR_UNIT_OF_TIME_YEAR_SUFFIX))
+        {
+            propertyFieldName = field.substring(field.startsWith("@") ? 1 : 0, field.length() - FIELD_SOLR_UNIT_OF_TIME_YEAR_SUFFIX.length());
+            ending = FIELD_SOLR_UNIT_OF_TIME_YEAR_SUFFIX;
+        }
```

Comment:
Probably an initial if with `field.contains("_unit_of_time_")` or so, encapsulating all the different cases could save some execution time.